In [20]:
import math

a,b,b1,x,mtriSup=[],[],[],[],[]

a=[[10,2,-1],[-3,-6,2],[1,1,5]]
b=[27,-61.5,-21.5]

def inicializar_Matriz(fil,col):
  
    mat= [None] * fil
    
    if(fil>0):
        for i in range(fil):
            mat[i] = [None] * col
            
    return mat

def inicilizar_Matriz_tri_sup(fil,col):
  
    mat= [None] * fil
    
    if(fil>0):
        for i in range(fil):
            mat[i] = [None] * col
            col-=1
            
    return mat
    
def copiar_matriz(a):
    
    mat=inicializar_Matriz(len(a),0);
    i,j=0,0
    
    while i<len(a):
        while j<len(a[i]):
            mat[i].append(a[i][j])
            j+=1
        j=0
        i+=1
    
    return mat

def imprimir_matriz(a):
    
    j=0
    
    while j<len(a):
        print(a[j])
        j+=1

def eliminacion_hacia_adelante(a,b):
    
    global b1,mtriSup
    i,x,z,a_,b1,mat=0,0,0,0,b[:],copiar_matriz(a)
    
    mtriSup=inicilizar_Matriz_tri_sup(len(a),len(a[0]))
    
    while z<len(mat):
        x=z
        while x<len(mat):
            if(z==x):
                a_=mat[x][x]
                b1[x]=b1[x]/a_
                while i<len(mat):
                    mat[z][i]=mat[z][i]/a_
                    i+=1
                i=0
            else:
                a_=mat[x][z]
                b1[x]=b1[x]-(b1[z]*a_)
                while i<len(mat):
                    mat[x][i]=mat[x][i]-(a_*mat[z][i])
                    i+=1
                i=0
            x+=1
        z+=1
        
    i,j,x=0,0,0
    
    while i<len(mat):
        while j<len(mat[i]):
            if(mat[i][j]!=0):
                mtriSup[i][x]=mat[i][j]
                x+=1
            j+=1
        j=0
        x=0
        i+=1

    print(mtriSup,b1)
    
def sustitucion_hacia_atras(mtriSup):
    
    global x
    i,j,x,z,bol=len(mtriSup)-1,len(mtriSup)-1,[0]*len(mtriSup),len(b1)-1,False
    
    while i>=0:
        aux=b1[i]
        j=len(mtriSup[i])-1
        while j>=0:
            if(mtriSup[i][j]!=1):
                aux-=mtriSup[i][j]*x[z]
                z-=1
            j-= 1

        x[i]=aux
        z=len(b1)-1
        i-=1
    i=0
        
    while i<len(x):
        print("X"+str(i+1)+"=",x[i])
        i+=1
                
    print(x)

eliminacion_hacia_adelante(a,b)
sustitucion_hacia_atras(mtriSup)


[[1.0, 0.2, -0.1], [1.0, -0.31481481481481477], [1.0]] [2.7, 9.888888888888888, -6.0]
X1= 0.5000000000000002
X2= 7.999999999999999
X3= -6.0
[0.5000000000000002, 7.999999999999999, -6.0]


Dadas las ecuaciones siguientes

10x1 + 2x2 – x3 = 27
–3x1 – 6x2 + 2x3 = –61.5
x1 + x2 + 5x3 = –21.5

a) Resuelva por eliminación de Gauss simple. Efectúe todos
los pasos del cálculo.
b) Sustituya los resultados en las ecuaciones originales a fin
de comprobar sus respuestas.

Solucion
a)
X1= 0.5000000000000002
X2= 7.999999999999999
X3= -6.0

10x1 + 2x2 – x3 = 27
–3x1 – 6x2 + 2x3 = –61.5
x1 + x2 + 5x3 = –21.5